src/core/ext/filters/client_channel/resolver/polling_resolver.cc

```diff
@@ -121,10 +121,11 @@ void PollingResolver::OnNextResolutionLocked() {
             "[polling resolver %p] re-resolution timer fired: shutdown_=%d",
             this, shutdown_);
   }
-  next_resolution_timer_handle_.reset();
-  if (!shutdown_) {
+  // If we haven't been cancelled nor shutdown, then start resolving.
```

Comment:
I don't think this PR is needed.

There are two places where the timer may be cancelled from:
- `ResetBackoffLocked()`: In this case, we want to abort the backoff delay and immediately re-resolve, so we *do* want to invoke `StartResolvingLocked()`.
- `ShutdownLocked()`: In this case, we do *not* want to invoke `StartResolvingLocked()`, but the existing code is already not doing that, because `shutdown_` is true.

Reply:
In the original `void PollingResolver::OnNextResolutionLocked(grpc_error_handle error)` function before the timer migration in #31717, this check was written like:
```
if (error.ok() && !shutdown_) {
  StartResolvingLocked();
}
```
See https://github.com/grpc/grpc/pull/31717/files#diff-beb9a20cca271b8a75724fca2a6587a6d0c3e2f79d856169b358419a057c6fcdL122.

So the logic was if the timer has not been cancelled and we are not shutting down, invoke `StartResolvingLocked()`. Maybe that is not intended in the first place?

Reply:
Done. Thanks for the great review Mark! Very clear! And it leaves a great record for future reference as well!